execution/block-partitioner/src/v2/partition_to_matrix.rs

```diff
@@ -0,0 +1,224 @@
+// Copyright © Aptos Foundation
+
+use crate::v2::{
+    counters::MISC_TIMERS_SECONDS,
+    extract_and_sort,
+    state::PartitionState,
+    types::{PrePartitionedTxnIdx, SenderIdx},
+    PartitionerV2,
+};
+use aptos_logger::trace;
+use aptos_types::block_executor::partitioner::{RoundId, TxnIndex};
+use dashmap::DashMap;
+use rayon::{
+    iter::ParallelIterator,
+    prelude::{IntoParallelIterator, IntoParallelRefIterator},
+};
+use std::{
+    mem,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        RwLock,
+    },
+};
+
+impl PartitionerV2 {
+    /// Populate `state.finalized_txn_matrix` with txns flattened into a matrix (num_rounds by num_shards),
+    /// in a way that avoid in-round cross-shard conflicts.
+    pub(crate) fn remove_cross_shard_dependencies(state: &mut PartitionState) {
+        let _timer = MISC_TIMERS_SECONDS
+            .with_label_values(&["remove_cross_shard_dependencies"])
+            .start_timer();
+
+        let mut remaining_txns = mem::take(&mut state.pre_partitioned);
```

Comment:
Why are we copying all the txns and then reseting the state ? Why not directly work on the mutable ref 'state.pre_partitioned' ?

Reply:
I think there's no copy, only object ownership transfer. So already equivalent to working on the mutable ref?

Reply:
ok got it